Block/Product.php

```diff
@@ -111,13 +111,19 @@ public function getProductSku() {
      *
      * @return bool|\Magento\Catalog\Model\Product
      */
-    public function getProduct()
-    {
-        if ($this->_coreRegistry->registry('product')) {
-            return $this->_coreRegistry->registry('product');
+    public function getProduct() {
+        if ( empty( $this->_product ) ) {
+            try {
+                $product        = $this->_coreRegistry->registry( 'product' );
+                $this->_product = $this->_productRepo->getById( $product->getId() );
```

Comment:
This is giving error of Call to a member function getId() on null in vendor/bazaarvoice/bazaarvoice-magento2-ext/Block/Product.php:118
we have product added on the home page using banner and widget